src/Providers/DTO/ProviderMetadata.php

```diff
@@ -60,13 +68,24 @@ class ProviderMetadata extends AbstractDataTransferObject
      * @param string $name The provider's display name.
      * @param ProviderTypeEnum $type The provider type.
      * @param string|null $credentialsUrl The URL where users can get credentials.
+     * @param RequestAuthenticationMethod|null $authenticationMethod The authentication method.
      */
-    public function __construct(string $id, string $name, ProviderTypeEnum $type, ?string $credentialsUrl = null)
-    {
+    public function __construct(
+        string $id,
+        string $name,
+        ProviderTypeEnum $type,
+        ?string $credentialsUrl = null,
+        ?RequestAuthenticationMethod $authenticationMethod = null
+    ) {
         $this->id = $id;
         $this->name = $name;
         $this->type = $type;
         $this->credentialsUrl = $credentialsUrl;
+
+        if ($authenticationMethod === null && $type->isCloud()) {
+            $authenticationMethod = RequestAuthenticationMethod::apiKey();
+        }
```

Comment:
Are we just considering this a safe assumption?

Reply:
Fair point. I added this for backward compatibility, but I already started thinking yesterday "what if there's a cloud provider that doesn't need any authentication and therefore _wants_ this to be `null`?

Maybe we just get rid of it and always use `null` as a default? It's a BC break, but probably not that important given almost nobody is implementing providers.